controllers/managedocs_controller.go

```diff
@@ -1338,6 +1338,25 @@ func (r *ManagedOCSReconciler) reconcileMonitoringResources() error {
 	for i := range promRuleList.Items {
 		obj := promRuleList.Items[i]
 		utils.AddLabel(obj, monLabelKey, monLabelValue)
+		// add an annotation "managed: true" to the prometheus rule with the label "prometheus: rook-prometheus"
+		if v, ok := obj.GetLabels()["prometheus"]; ok && v == "rook-prometheus" {
+			// remove unwanted fields from the prometheus rule
+			unwantedRuleAnnotations := []string{"documentation"}
+			groups := obj.Spec.Groups
+			for groupIndex := range groups {
+				group := &groups[groupIndex]
+				for ruleIndex := range group.Rules {
+					rule := &group.Rules[ruleIndex]
+					if rule.Alert != "" {
+						for _, annotation := range unwantedRuleAnnotations {
+							delete(rule.Annotations, annotation)
+						}
+					}
+				}
+			}
```

Comment:
1. What component is setting this annotation in the first place?
2. What happens if we don't remove this annotation, does the side effect of not removing this annotation justify the bulk of reconcile code here for removing this single annotation?

Reply:
1. The `documentation` annotation will be set in the OCS Ceph rules by default, following the Ceph rule format: https://github.com/ceph/ceph/blob/main/monitoring/ceph-mixin/prometheus_alerts.yml#L28.
2. Yep, if this is not removed, the receivers will send out notifications to managed services users having this links in them, which is something we absolutely want to avoid.

Reply:
- ack, thanks. In that scenario, I believe relabelling the alert is more natural rather than editing the rule itself 🤔
- I do understand that in the linked PR you are trying not to reconcile specific annotated rules in the product, however even for verifying the absence of that annotation we need to drill through multiple stanza's which doesn't sound optimal
- Either come up with a relabel rule just before sending out the alerts or save the state that you removed the annotation and product will not reconcile that
- Nevertheless, I'd like to wait for others to weigh in their thoughts

Reply:
Yeah, relabelling should work as well. +1.

> Either come up with a relabel rule just before sending out the alerts or save the state that you removed the annotation and product will not reconcile that

Makes sense.